### PR TITLE
doc: add auto_id_cache table option explanation

### PR DESCRIPTION
**Pull request by AilinKid**
### What is changed, added or deleted? (Required)
add auto_id_cache table option explanation

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/mysql-compatibility.md [hunk 1]
```diff
@@ -85,6 +85,10 @@ mysql> select _tidb_rowid, id from t;
 3 rows in set (0.01 sec)
 ```
 
+TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的，其会随着上次分配缓存大小的使用速度，进行下一次分配缓存大小的调整，整体范围在 30000 到 2000000 内进行浮动。在 v4.0.rc.2, v3.1.2 和 v3.0.14 版本中，TiDB 引入了 `AUTO_ID_CACHE` table option 来允许用户自主设置自增 ID 分配缓存的大小。
+
+由于该参数是作用在自增列分配器（可能由 auto_increment 和 _tidb_rowid 共用）的逻辑上，因此在该缓存大小的使用可能会被 auto_increment 和 _tidb_rowid 共同消耗。此外如果在 insert 语句中所需连续 id 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当放大缓存步长以便能够保证该语句的正常插入。
+
```
Comment (djshow832) [inline reference/mysql-compatibility.md:91]:
```suggestion
TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的。从 v3.1.2, v3.0.14 和 v4.0.rc.2 版本开始，TiDB 引入了 `AUTO_ID_CACHE` 表选项来允许用户自主设置自增 ID 分配缓存的大小。其中缓存大小的使用可能会被 auto_increment 和 `_tidb_rowid` 共同消耗。此外如果在 INSERT 语句中所需连续 ID 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当调大缓存大小以便能够保证该语句的正常插入。

```

It's too detailed, which may make users confused.

Also, please add an example here, or put a link of the usage of `AUTO_ID_CACHE`.

File: reference/mysql-compatibility.md [hunk 2]
```diff
@@ -85,6 +85,10 @@ mysql> select _tidb_rowid, id from t;
 3 rows in set (0.01 sec)
 ```
 
+TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的，其会随着上次分配缓存大小的使用速度，进行下一次分配缓存大小的调整，整体范围在 30000 到 2000000 内进行浮动。在 v4.0.rc.2, v3.1.2 和 v3.0.14 版本中，TiDB 引入了 `AUTO_ID_CACHE` table option 来允许用户自主设置自增 ID 分配缓存的大小。
+
+由于该参数是作用在自增列分配器（可能由 auto_increment 和 _tidb_rowid 共用）的逻辑上，因此在该缓存大小的使用可能会被 auto_increment 和 _tidb_rowid 共同消耗。此外如果在 insert 语句中所需连续 id 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当放大缓存步长以便能够保证该语句的正常插入。
```
Comment (TomShawn) [inline reference/mysql-compatibility.md:90]:
```suggestion
由于该参数是作用在自增列分配器（可能由 `auto_increment` 和 `_tidb_rowid` 共用）的逻辑上，因此在该缓存大小的使用可能会被 `auto_increment` 和 `_tidb_rowid` 共同消耗。此外如果在 `insert` 语句中所需连续 ID 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当放大缓存步长以便能够保证该语句的正常插入。
```

File: reference/mysql-compatibility.md (hunk 1 above)
Comment (AilinKid) [reply reference/mysql-compatibility.md]: Usage？it was！

File: reference/mysql-compatibility.md [hunk 3]
```diff
@@ -85,6 +85,8 @@ mysql> select _tidb_rowid, id from t;
 3 rows in set (0.01 sec)
 ```
 
+TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的。从 v3.1.2, v3.0.14 和 v4.0.rc.2 版本开始，TiDB 引入了 `AUTO_ID_CACHE` 表选项来允许用户自主设置自增 ID 分配缓存的大小。其中缓存大小的使用可能会被 auto_increment 和 `_tidb_rowid` 共同消耗。此外如果在 INSERT 语句中所需连续 ID 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当调大缓存大小以便能够保证该语句的正常插入。
```
Comment (djshow832) [inline reference/mysql-compatibility.md:88]:
```suggestion
TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的。从 v3.1.2, v3.0.14 和 v4.0.rc.2 版本开始，TiDB 引入了 `AUTO_ID_CACHE` 表选项来允许用户自主设置自增 ID 分配缓存的大小。其中缓存大小的使用可能会被自增列和 `_tidb_rowid` 共同消耗。此外如果在 INSERT 语句中所需连续 ID 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当调大缓存以便能够保证该语句的正常插入。
```

Comment (TomShawn) [inline reference/mysql-compatibility.md:88]: `v3.1.2` 和 `v4.0.0-rc.2` 都是没有发布的版本吧。
```suggestion
TiDB 自增 ID 的缓存大小在早期版本中是对用户透明的。从 v3.1.2、v3.0.14 和 v4.0.rc.2 版本开始，TiDB 引入了 `AUTO_ID_CACHE` 表选项来允许用户自主设置自增 ID 分配缓存的大小。其中缓存大小可能会被 `auto_increment` 和 `_tidb_rowid` 共同消耗。此外如果在 `INSERT` 语句中所需连续 ID 长度超过 `AUTO_ID_CACHE` 的长度时，TiDB 会适当调大缓存大小以便能够保证该语句的正常插入。
```

Comment (AilinKid) [reply reference/mysql-compatibility.md:88]: 是的

**Comment by TomShawn** [comment]
LGTM

**Comment by AilinKid** [comment]
> LGTM
> But do not merge now, because the corresponding versions are not published yet.

zimuxia think it's ok, cause it will come soon~ @djshow832

**Pull request merged**